shanoir-ng-datasets/src/main/java/org/shanoir/ng/vip/execution/service/ExecutionTrackingServiceImpl.java

```diff
@@ -0,0 +1,181 @@
+
+@Service
+public class ExecutionTrackingServiceImpl implements ExecutionTrackingService {
+
+    @Value("${vip-data-folder}")
+    private String trackingFilePrefixe;
+
+    private int MAX_LAST_LINES_TO_CHECK = 10;
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExecutionTrackingServiceImpl.class);
+
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("HH:mm dd/MM/yyyy");
+
+    public enum execStatus {VALID, SENT}
+
+    public void updateTrackingFile(ExecutionMonitoring executionMonitoring, execStatus execStatus) {
+        try {
+            File trackingFile = new File(getTrackingFilePath(executionMonitoring));
+            createTrackingFile(trackingFile);
+
+            switch (execStatus) {
+                case VALID -> createTrackingLine(executionMonitoring, trackingFile);
+                case SENT -> updateTrackingLine(executionMonitoring, trackingFile);
+                default -> throw new ShanoirException("Invalid execution status, can not track.");
+            }
+        } catch (IOException | ShanoirException e) {
+            LOG.error("An error occured while trying to write in VIP tracking file", e);
+        }
+    }
+
+    public void completeTracking(ExecutionMonitoring executionMonitoring, DatasetProcessing newProcessing) {
+        try {
+            File trackingFile = new File(getTrackingFilePath(executionMonitoring));
+            List<String> lastLines = getLastLines(trackingFile);
+
+            boolean retrievedLine = false;
+            for (String line : lastLines) {
+                List<String> lineParts = new ArrayList<>(Arrays.asList(line.split(",")));
+
+                if(Long.parseLong(lineParts.get(1)) == executionMonitoring.getId()) {
+                    lineParts.set(1, newProcessing.getId().toString());
+                    lineParts.add(newProcessing.getOutputDatasets().stream().anyMatch(file -> Objects.equals("error.yaml", file.getName())) ? "true" : "false");
+                    lineParts.add(newProcessing.getOutputDatasets().stream().anyMatch(file -> Objects.equals("results.yaml", file.getName())) ? "true" : "false");
+
+                    lastLines.set(lastLines.indexOf(line), String.join(",", lineParts));
+                    retrievedLine = true;
+                    break;
+                }
+            }
+            if (!retrievedLine) {
+                throw new ShanoirException("Execution monitoring tracking line is lost, can not complete line.");
+            }
+
+            writeLastLines(lastLines, trackingFile);
+        } catch (IOException | ShanoirException e) {
+            LOG.error("An error occured while trying to write in VIP tracking file", e);
+        }
+    }
+
+    /**
+     * Create a new line for the execution input
+     */
+    private void createTrackingLine(ExecutionMonitoring executionMonitoring, File trackingFile) throws IOException {
+        BufferedWriter writer = new BufferedWriter(new FileWriter(trackingFile,true));
```

Comment:
Hi @DuckflipXYZ, thank you for this great PR.
I think it would be nice to use a try(resource) statement here to close the writer in any possible case?